core/src/main/java/com/datastax/oss/driver/internal/core/config/cloud/CloudConfigFactory.java

```diff
@@ -225,22 +228,40 @@ protected TrustManagerFactory createTrustManagerFactory(
   @NonNull
   protected BufferedReader fetchProxyMetadata(
       @NonNull URL metadataServiceUrl, @NonNull SSLContext sslContext) throws IOException {
-    try {
-      HttpsURLConnection connection = (HttpsURLConnection) metadataServiceUrl.openConnection();
-      connection.setSSLSocketFactory(sslContext.getSocketFactory());
-      connection.setRequestMethod("GET");
-      connection.setRequestProperty("host", "localhost");
-      return new BufferedReader(
-          new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
-    } catch (ConnectException e) {
-      throw new IllegalStateException(
-          "Unable to connect to cloud metadata service. Please make sure your cluster is not parked or terminated",
-          e);
-    } catch (UnknownHostException e) {
-      throw new IllegalStateException(
-          "Unable to resolve host for cloud metadata service. Please make sure your cluster is not terminated",
-          e);
+    HttpsURLConnection connection = null;
+    int attempt = 0;
+    while(attempt < METADATA_RETRY_MAX_ATTEMPTS){
+      try {
+        connection = (HttpsURLConnection) metadataServiceUrl.openConnection();
+        connection.setSSLSocketFactory(sslContext.getSocketFactory());
+        connection.setRequestMethod("GET");
+        connection.setRequestProperty("host", "localhost");
+        attempt++;
```

Comment:
It's not a problem _logically_ but I'd argue it's a bit cleaner to increment the number of attempts at the very top of this loop.  You can probably do it even before going into the try block honestly.